tlr/stages-tlr/model-provider/src/main/java/edu/kit/kastel/mcse/ardoco/tlr/models/informants/LargeLanguageModel.java

```diff
@@ -0,0 +1,104 @@
+/* Licensed under MIT 2024-2025. */
+package edu.kit.kastel.mcse.ardoco.tlr.models.informants;
+
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import dev.langchain4j.model.chat.ChatModel;
+import dev.langchain4j.model.ollama.OllamaChatModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import edu.kit.kastel.mcse.ardoco.core.architecture.Deterministic;
+import edu.kit.kastel.mcse.ardoco.core.common.util.Environment;
+
+@Deterministic
+public enum LargeLanguageModel {
+    // OPENAI
+    GPT_4_O_MINI("GPT-4o mini", () -> createOpenAiModel("gpt-4o-mini-2024-07-18")), //
+    GPT_4_O("GPT-4o", () -> createOpenAiModel("gpt-4o-2024-08-06")), //
+    GPT_4_TURBO("GPT-4 Turbo", () -> createOpenAiModel("gpt-4-turbo-2024-04-09")), //
+    GPT_4("GPT-4", () -> createOpenAiModel("gpt-4-0613")), //
+    GPT_3_5_TURBO("GPT-3.5 Turbo", () -> createOpenAiModel("gpt-3.5-turbo-0125")), //
+    OPENAI_GENERIC(Environment.getenv("OPENAI_MODEL_NAME"), () -> createOpenAiModel(Environment.getenv("OPENAI_MODEL_NAME"))), //
+    // OLLAMA
+    CODELLAMA_13B("Codellama 13b", () -> createOllamaModel("codellama:13b")), //
+    // CODELLAMA_70B("Codellama 70b", () -> createOllamaModel("codellama:70b")), //
+    //
+    // GEMMA_2_27B("Gemma2 27b", () -> createOllamaModel("gemma2:27b")), //
+    //
+    // QWEN_2_72B("Qwen2 72b", () -> createOllamaModel("qwen2:72b")), //
+    //
+    LLAMA_3_1_8B("Llama3.1 8b", () -> createOllamaModel("llama3.1:8b-instruct-fp16")), //
+    LLAMA_3_1_70B("Llama3.1 70b", () -> createOllamaModel("llama3.1:70b")), //
+    //
+    // MISTRAL_7B("Mistral 7b", () -> createOllamaModel("mistral:7b")), //
+    // MISTRAL_NEMO_27B("Mistral Nemo 12b", () -> createOllamaModel("mistral-nemo:12b")), //
+    // MIXTRAL_8_X_22B("Mixtral 8x22b", () -> createOllamaModel("mixtral:8x22b")), //
+    //
+    // PHI_3_14B("Phi3 14b", () -> createOllamaModel("phi3:14b")), //
+    //
```

Comment:
Why are the values commented? Please decide on some models. Either add or remove them